operator-framework/src/test/java/com/github/containersolutions/operator/sample/subresource/SubResourceTestCustomResourceController.java

```diff
@@ -0,0 +1,50 @@
+package com.github.containersolutions.operator.sample.subresource;
+
+import com.github.containersolutions.operator.TestExecutionInfoProvider;
+import com.github.containersolutions.operator.api.Context;
+import com.github.containersolutions.operator.api.Controller;
+import com.github.containersolutions.operator.api.ResourceController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Controller(
+        crdName = SubResourceTestCustomResourceController.CRD_NAME,
+        customResourceClass = SubResourceTestCustomResource.class)
+public class SubResourceTestCustomResourceController implements ResourceController<SubResourceTestCustomResource>,
+        TestExecutionInfoProvider {
+
+    public static final String CRD_NAME = "subresourcesample.sample.javaoperatorsdk";
+    private static final Logger log = LoggerFactory.getLogger(SubResourceTestCustomResourceController.class);
+    private AtomicInteger numberOfExecutions = new AtomicInteger(0);
+
+    @Override
+    public boolean deleteResource(SubResourceTestCustomResource resource, Context<SubResourceTestCustomResource> context) {
+        return true;
+    }
+
+    @Override
+    public Optional<SubResourceTestCustomResource> createOrUpdateResource(SubResourceTestCustomResource resource, Context<SubResourceTestCustomResource> context) {
+        numberOfExecutions.addAndGet(1);
+        log.info("Value: " + resource.getSpec().getValue());
+
+        ensureStatusExists(resource);
+        resource.getStatus().setState(SubResourceTestCustomResourceStatus.State.SUCCESS);
+        context.getCustomResourceClient().updateStatus(resource);
```

Comment:
I don't understand this context-based approach. The controller has to return the CR to tell the framework to update it, but it also has to update it using the context?

Reply:
Yes, the call for the subresource is an another request. You cannot update it by updating the resourc or vice versa if marked as subresource in crd.

We could design a nicer api, but by definition you can have more subresource types ( although not really supported this by fabric8 ). So for now this is a quick and general solution.

Reply:
By returning the cr we can update anything except subresource. This is by design of k8s.

Reply:
What I could think as an alternative, is just return the CR as we do now, and a flag if we should also call the update status. But still we want in any circumstances the context, so we have the CR client available.